transform-plugins/src/main/java/co/cask/hydrator/plugin/Encoder.java

```diff
@@ -0,0 +1,252 @@
+    super.initialize(context);
+    parseConfiguration(config.encode);
+    try {
+      outSchema = Schema.parseJson(config.schema);
+      List<Field> outFields = outSchema.getFields();
+      for (Field field : outFields) {
+        outSchemaMap.put(field.getName(), field.getSchema().getType());
+      }
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Format of schema specified is invalid. Please check the format.");
+    }
+  }
+
+  @Override
+  public void transform(StructuredRecord in, Emitter<StructuredRecord> emitter) throws Exception {
+    StructuredRecord.Builder builder = StructuredRecord.builder(outSchema);
+
+    Schema inSchema = in.getSchema();
+    List<Field> inFields = inSchema.getFields();
+
+    // Iterate through input fields. Check if field name is present 
+    // in the fields that need to be encoded, if it's not then write 
+    // to output as it is. 
+    for (Field field : inFields) {
+      String name = field.getName();
+
+      // Check if output schema also have the same field name. If it's not 
+      // then throw an exception. 
+      if (!outSchemaMap.containsKey(name)) {
+        continue;
+      }
+
+      Schema.Type outFieldType = outSchemaMap.get(name);
+
+      // Check if the input field name is configured to be encoded. If the field is not 
+      // present or is defined as none, then pass through the field as is. 
+      if (!encodeMap.containsKey(name) || encodeMap.get(name) == EncodeType.NONE) {
+        builder.set(name, in.get(name));
+      } else {
+        // Now, the input field could be of type String or byte[], so transform everything
+        // to byte[] 
+        byte[] obj = new byte[0];
+        if (field.getSchema().getType() == Schema.Type.STRING) {
+          obj = ((String) in.get(name)).getBytes();
+        } else if (field.getSchema().getType() == Schema.Type.BYTES) {
+          obj = in.get(name);
+        }
+
+        // Now, based on the encode type configured for the field - encode the byte[] of the 
+        // value.
+        byte[] outValue = new byte[0];
+        EncodeType type = encodeMap.get(name);
+        if (type == EncodeType.STRING_BASE32) {
+          outValue = base32Encoder.encodeAsString(obj).getBytes();
+        } else if (type == EncodeType.BASE32) {
+          outValue = base32Encoder.encode(obj);
+        } else if (type == EncodeType.STRING_BASE64) {
+          outValue = base64Encoder.encodeAsString(obj).getBytes();
+        } else if (type == EncodeType.BASE64) {
+          outValue = base64Encoder.encode(obj);
+        } else if (type == EncodeType.HEX) {
+          outValue = hexEncoder.encode(obj);
+        }
+
+        // Depending on the output field type, either convert it to 
+        // Bytes or to String. 
+        if (outFieldType == Schema.Type.BYTES) {
+          builder.set(name, outValue);
+        } else if (outFieldType == Schema.Type.STRING) {
+          builder.set(name, new String(outValue));
+        }
```

Comment:
The issue is it should not happen and it should be caught in configure (which we cannot today as we don't know the schema of input). But, let's say we throw here then for every record we will emit a log line. It's not a good experience. It's a lot of noise.